### PR TITLE
Feature/add secc configurable parameters

### PR DESCRIPTION
**Pull request by aadritG**
Feature added to allow usage of configuration files (.yaml) passed as argument for setting the SECC configuration parameters for the charging session. This allows the tester to keep separate SECC configurations for distinct testing scenarios, avoiding the need to change the source code each time.

example:
$ make  run-secc  config=conf_default.yaml

If no argument is passed. default configurations shall be selected from conf_default.yaml

The configuration file is read in iso15118/secc/main.py and the returned dictionary is passed to SimEVSEController.create() and config.load_envs(). During the charging session, If the parameters in the config file are existent and valid, it is loaded, else the default fallback value is loaded.

Note: 
Might need to do the following to install pyyaml,
$poetry  add  pyyaml

Additionally, the protocol list in secc_settings has been sorted.

- Aadrit Ghimire, Lucid Motors

### CODE REVIEW

**Comment by tropxy** [comment]
hi, thanks for your contribution, could you rebase your branch?

**Comment by aadritG** [comment]
Rebased

File: iso15118/secc/controller/simulator.py [hunk 1]
```diff
@@ -908,3 +953,15 @@ async def get_15118_ev_certificate(
         ).decode("utf-8")
 
         return base64_encode_cert_install_res
+
+    def get_cfg_parameter(self, string, default):
+        try:
+                if self.config_param[string] is None:                      
+                        raise Exception("revert to default")
```
Comment (tropxy) [inline iso15118/secc/controller/simulator.py:960]: there is no point on raising a general exception here. log as info and return the default

File: iso15118/secc/controller/simulator.py [hunk 2]
```diff
@@ -908,3 +953,15 @@ async def get_15118_ev_certificate(
...
+                logger.error(f"No valid configuration for {string}, revert to default")
```
Comment (tropxy) [inline iso15118/secc/controller/simulator.py:961]: the message should be in the same indentation level as the raise above, nevertheless as I said I think this should be a INFO level and not an ERROR one

File: iso15118/secc/main.py [hunk 3]
```diff
@@ -1,21 +1,39 @@
...
-    sim_evse_controller = await SimEVSEController.create()
+
+    global args
+
+    default_config_file = "conf_default.yaml"
+
+    try:
+        with open(args[1]) as f:
+            config_param = yaml.load(f, Loader=yaml.FullLoader)
+        print("Loading configuration parameters from " + args[1])
```
Comment (tropxy) [inline iso15118/secc/main.py:27]: you can remove this line

File: iso15118/secc/main.py [hunk 4]
```diff
@@ -1,21 +1,39 @@
...
+        logger.info(f"Loaded configuration parameters from: {args[1]}")
+    except:
```
Comment (tropxy) [inline iso15118/secc/main.py:29]: what is the specific exception that is raised? I would assume "FileNotFoundError"?

**Review by tropxy — CHANGES_REQUESTED** [review]

File: iso15118/secc/controller/simulator.py (hunk 1 above)
Comment (aadritG) [reply iso15118/secc/controller/simulator.py]: Updated! Thanks for the suggestion

File: iso15118/secc/controller/simulator.py (hunk 2 above)
Comment (aadritG) [reply iso15118/secc/controller/simulator.py]: Updated as specified above

File: iso15118/secc/main.py (hunk 4 above)
Comment (aadritG) [reply iso15118/secc/main.py:28]: Yes. Additionally, if no configfile is passed as argument, line 25 will result in an IndexError

> with open(args[1]) as f:

**Pull request closed without merging**